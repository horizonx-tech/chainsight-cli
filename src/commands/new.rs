use clap::Parser;

#[derive(Debug, Parser)]
#[command(name = "new")]
/// Generate ChainSight project by prepared template
pub struct NewOpts {}