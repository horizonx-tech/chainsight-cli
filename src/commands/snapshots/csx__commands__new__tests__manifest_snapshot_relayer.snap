---
source: src/commands/new.rs
expression: template_relayer_manifest(COMPONENT_PREFIX).to_str_as_yaml().unwrap()
---
# yaml-language-server: $schema=https://raw.githubusercontent.com/horizonx-tech/chainsight-cli/main/resources/schema/relayer.json
version: v1
metadata:
  label: sample_relayer
  type: relayer
  description: ''
  tags:
  - Oracle
  - snapshot
datasource:
  type: canister
  location:
    id: sample_snapshot_indexer_chain
    args:
      id_type: canister_name
  method:
    identifier: 'get_last_snapshot_value : () -> (text)'
    interface: null
    args: []
destination:
  network_id: 80001
  type: uint256
  oracle_address: 0539a0EF8e5E60891fFf0958A059E049e43020d9
  rpc_url: https://polygon-mumbai.g.alchemy.com/v2/<YOUR_KEY>
interval: 3600
lens_targets: null

